File: python/armstrong_numbers.py
# How can you make this more scalable and reusable later?
def find_armstrong_numbers(numbers):
    answer = []
    sum_of_array = 0
   
    for value in numbers:
        if value < 10:
            answer.append(value)
        else:
            initial_sum = 0
            split_number = []
            string_value = str(value)
            for number in range(0, len(string_value)):
                split_number.append(int(string_value[number]))
            for secondNumber in split_number:
                initial_sum += secondNumber ** len(split_number)
            sum_of_array = initial_sum
            if sum_of_array == value:
             answer.append(sum_of_array)
    return answer  
   
        

File: js/armstrongNumbers.js
// How can you make this more scalable and reusable later?

//exports.

findArmstrongNumbers = function(arr) {

let answer = [];
let sumOfArr = 0;

// let splitNum = arr.toString().split('').map(elem => (elem));

for (let i = 0; i <= arr[0]; i++ )
{
    if (i < 10)
    {
        answer.push(i);
    }else{
        let splitNum = i.toString().split('').map(elem => (elem));
        let initialSum = 0;
    for (let j = 0; j < splitNum.length; j++)
    {
        initialSum += splitNum[j] ** splitNum.length;
        
    }
    sumOfArr = initialSum;
    if (sumOfArr == parseInt(splitNum.join(""))){
        answer.push(sumOfArr);
    }
    }
}return answer

};
console.log(findArmstrongNumbers([8]))
console.log(findArmstrongNumbers([99]))
console.log(findArmstrongNumbers([999]))